src/diffusers/models/attention_processor.py

```diff
@@ -369,7 +369,20 @@ def set_use_memory_efficient_attention_xformers(
                 )
                 processor = XFormersAttnAddedKVProcessor(attention_op=attention_op)
             else:
-                processor = XFormersAttnProcessor(attention_op=attention_op)
+                processor = self.processor
+                if isinstance(self.processor, (IPAdapterAttnProcessor, IPAdapterAttnProcessor2_0)):
```

Comment:
let's add a `is_ip_adapter` flag similar to is_custom_diffusion etc

```python
is_ip_adapter = hasattr(self, "processor") and isinstance(
            self.processor,( IPAdapterAttnProcessor, IPAdapterAttnProcessor2_0),
        )
```

Reply:
Yes, it is perfectly possible, but it will have to be like below, so that the modules that have already been changed to the Xformers attention class are not replaced again to the XFormersAttnProcessor class in the final Else.

```python
is_ip_adapter = hasattr(self, "processor") and isinstance(
            self.processor, 
            (IPAdapterAttnProcessor, IPAdapterAttnProcessor2_0, IPAdapterXFormersAttnProcessor),
        )

Reply:
yes, the code you show here is ok! 
we just want to keep a consistent style that's all:)